src/index.ts

```diff
@@ -1 +1,535 @@
+                            `Cannot start new traffic collection. Previous one "${inProgressTrafficCollectionName}" is still running.`
+                        );
+                    if (analysisOptions.platform === 'ios') throw new Error('Unimplemented'); // TODO: We don't have wireguard support on iOS, yet.
+
+                    platform.installCertificateAuthority(join(homedir(), '.mitmproxy/mitmproxy-ca-cert.pem'));
+
+                    inProgressTrafficCollectionName = name ?? new Date().toISOString();
+
+                    const harOutputPath = temporaryFile({ extension: 'har' });
+
+                    mitmproxyState = {
+                        proc: execa(
+                            'mitmdump',
+                            [
+                                '--mode',
+                                'wireguard',
+                                ...Object.entries(mitmproxyAddons).map(([addonName, addonPath]) => {
+                                    if (!existsSync(addonPath))
+                                        throw new Error(
+                                            `No "${addonName}" addon for mitmproxy found at "${addonPath}".${
+                                                addonName === 'ipcEvents' || addonName === 'harDump'
+                                                    ? ' The mitmproxy capability requires that this addon is present.'
+                                                    : ''
+                                            }`
+                                        );
+                                    return `-s ${addonPath}`;
+                                }),
+                                '--set',
+                                `hardump=${harOutputPath}`,
+                                '--set',
+                                'ipcPipeFd=3',
+                            ],
+                            {
+                                stdio: ['pipe', 'pipe', 'pipe', 'ipc'],
+                            }
+                        ),
+                        harOutputPath,
+                    };
+
+                    await timeout(
+                        Promise.all([
+                            awaitMitmproxyEvent(
+                                mitmproxyState.proc,
+                                (msg) =>
+                                    msg.status === 'proxyChanged' &&
+                                    msg.servers.some((server) => server.type === 'wireguard' && server.is_running)
+                            )
+                                .then((msg) => {
+                                    if (msg.status !== 'proxyChanged') throw new Error('Unreachable.'); // This will never be reached but we use it as a typeguard
+                                    for (const server of msg.servers) {
+                                        if (
+                                            (server.type !== 'wireguard' && !server.wireguard_conf) ||
```

Comment:
Shouldn't this be `server.type !== 'wireguard' || !server.wireguard_conf`. Having a `wireguard_conf` with a different type sounds like a state that should never be reached?

Reply:
That's a typeguard because it didn't think it would be worth it to create an even more complicated type for `MitmproxyEvent`.

Reply:
Sure, but that shouldn't prevent you from using `||`?

Reply:
Ah, but the `wireguard_conf` can be `null` even if `server.type = 'wireguard'`, which is the case when the server dind't fully start up, yet.